libs/vkd3d/vkd3d_private.h

```diff
@@ -1197,7 +1196,22 @@ struct vkd3d_descriptor_metadata_buffer_view
     uint32_t range;
     /* Format used if used in a formatted context such as UAV clears.
      * If UNKNOWN, denotes a raw buffer. R32_UINT can be used in place of it. */
-    DXGI_FORMAT dxgi_format;
+    uint16_t padding;
+    uint8_t dxgi_format; /* All valid formats fit in 8-bit. */
+    uint8_t flags; /* Important that this the last member. It is used as the generic flags field. */
```

Comment:
The problem is the forced padding I think. I'm not sure we can deal with the padding and stay within 16 byte.

Reply:
Or perhaps we can ... I'll fiddle with it.

Reply:
I meant something like keeping the union of structs but just moving flags to the top. This is doable without introducing extra padding since we'd just be reordering struct members.

Either way I don't think it's super important.

Reply:
Refactored layout to be less whack.